unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/Services/RestrictedActionsService/RestrictedActionsServiceImpl.cs

```diff
@@ -0,0 +1,120 @@
+using Cysharp.Threading.Tasks;
+using Decentraland.Renderer.RendererServices;
+using rpc_csharp;
+using RPC.Context;
+using System;
+using System.Threading;
+using UnityEngine;
+
+namespace RPC.Services
+{
+    public class RestrictedActionsServiceImpl : IRestrictedActionsService<RPCContext>
+    {
+        public const int MAX_ELAPSED_FRAMES_SINCE_INPUT = 30;
+
+        private static readonly SuccessResponse SUCCESS_RESPONSE = new SuccessResponse() { Success = true };
+        private static readonly SuccessResponse FAIL_RESPONSE = new SuccessResponse() { Success = false };
+
+        public static void RegisterService(RpcServerPort<RPCContext> port)
+        {
+            RestrictedActionsServiceCodeGen.RegisterService(port, new RestrictedActionsServiceImpl());
+        }
+
+        public async UniTask<SuccessResponse> OpenExternalUrl(OpenExternalUrlRequest request, RPCContext context, CancellationToken ct)
+        {
+            await UniTask.SwitchToMainThread(ct);
+            RestrictedActionsContext restrictedActions = context.restrictedActions;
+
+            bool success = false;
+            int currentFrameCount = restrictedActions.GetCurrentFrameCount?.Invoke() ?? GetCurrentFrameCount();
+
+            try
+            {
+                if ((currentFrameCount - restrictedActions.LastFrameWithInput) <= MAX_ELAPSED_FRAMES_SINCE_INPUT)
+                {
+                    success = restrictedActions.OpenExternalUrlPrompt?.Invoke(request.Url, request.SceneNumber) ?? false;
+                }
+            }
+            catch (Exception _)
+            { // ignored
+            }
+
+            return success ? SUCCESS_RESPONSE : FAIL_RESPONSE;
+        }
+
+        public async UniTask<SuccessResponse> OpenNftDialog(OpenNftDialogRequest request, RPCContext context, CancellationToken ct)
+        {
+            await UniTask.SwitchToMainThread(ct);
+            RestrictedActionsContext restrictedActions = context.restrictedActions;
+
+            bool success = false;
+            int currentFrameCount = restrictedActions.GetCurrentFrameCount?.Invoke() ?? GetCurrentFrameCount();
+
+            try
+            {
+                if ((currentFrameCount - restrictedActions.LastFrameWithInput) <= MAX_ELAPSED_FRAMES_SINCE_INPUT)
+                {
+                    if (TryParseUrn(request.Urn, out string contractAddress, out string tokenId))
+                    {
+                        restrictedActions.OpenNftPrompt?.Invoke(contractAddress, tokenId);
+                        success = true;
+                    }
+                }
+            }
+            catch (Exception _)
+            { // ignored
+            }
+
+            return success ? SUCCESS_RESPONSE : FAIL_RESPONSE;
+        }
+
+        private static int GetCurrentFrameCount()
+        {
+            return Time.frameCount;
+        }
```

Comment:
this is a temporal solution, I will change renderer's frame count for scene's tick later on another PR so we avoid having issues with renderer update vs scene update

Reply:
Can you add a comment with this on the code?